### PR TITLE
Add common-instancetypes repo to KubeVirt

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @ksimon1 

This project will hopefully eventually replace common-templates. It has been residing under my personal account for a while now and is ready to start the migration to the KubeVirt namespace.

### CODE REVIEW

File: github/ci/prow-deploy/files/orgs.yaml [hunk 1]
```diff
@@ -270,6 +270,9 @@ orgs:
         description: This is a demonstration of how KubeVirt could be displayed in Cockpit
         has_projects: false
         has_wiki: false
+      common-instancetypes:
```
Comment (ksimon1) [inline github/ci/prow-deploy/files/orgs.yaml:273]: I am wondering if name common-instance-types would be better :)

Comment (lyarwood) [reply github/ci/prow-deploy/files/orgs.yaml:273]: hmmm the term isn't hyphenated or even separated in our API or user docs, the latter might be a mistake on my part given AWS does refer to their version as `instance types`. I'm on the fence about the additional hyphen here tbh but I'll let others chime in.

Comment (dominikholler) [reply github/ci/prow-deploy/files/orgs.yaml:273]: I vote also for keeping all realted naming consistent with the API, which means using `common-instancetypes` here.

**Comment by lyarwood** [comment]
@dhiller @brianmcarey would either of you mind looking at this and helping me move it forward. I've essentially copied the approach taken in #2303 so please let me know if this is wrong.

**Review by brianmcarey — APPROVED** [review]
/approve

**Comment by fabiand** [comment]
FWIW - I'd even have considered to call the repo just `instancetypes`. The benefit of it would have been a pretty nice URL: github.com/kubevirt/instancetypes which is easy to remember and descriptive.
After all I'm expceting (hoping!) that this URL will be commonly used to create custom derivatives.

Worth to change it?

**Comment by lyarwood** [comment]
> FWIW - I'd even have considered to call the repo just `instancetypes`. The benefit of it would have been a pretty nice URL: github.com/kubevirt/instancetypes which is easy to remember and descriptive. After all I'm expceting (hoping!) that this URL will be commonly used to create custom derivatives.
> 
> Worth to change it?

To me this would suggest that these are the only instancetypes available instead of a suggested common set so I'd rather stick with the current naming.

**Pull request merged**